lib/server.js

```diff
@@ -36,39 +36,37 @@ function createStream(client) {
   return stream;
 }
 
+class ClientStream extends Duplex {
+  constructor(client) {
+    super({ objectMode: true });
 
-function ClientStream(client) {
-  Duplex.call(this, {objectMode: true});
+    this.client = client;
+    const self = this;
 
-  this.client = client;
+    this.on('error', function(error) {
+      console.warn('BrowserChannel client message stream error', error.stack || error);
+      self._stopClient();
+    });
 
-  const self = this;
+    // The server ended the writable stream. Triggered by calling stream.end()
+    // in agent.close()
+    this.on('finish', function() {
+      self._stopClient();
+    });
+  }
 
-  this.on('error', function(error) {
-    console.warn('BrowserChannel client message stream error', error.stack || error);
-    self._stopClient();
-  });
+  _read() {};
 
-  // The server ended the writable stream. Triggered by calling stream.end()
-  // in agent.close()
-  this.on('finish', function() {
-    self._stopClient();
-  });
-}
-util.inherits(ClientStream, Duplex);
+  _write(chunk, encoding, callback) {
+    // Silently drop messages after the session is closed
+    if (this.client.state === 'closed') return;
+    this.client.send(chunk);
+    callback();
+  };
 
-ClientStream.prototype._read = function() {};
-
-ClientStream.prototype._write = function(chunk, encoding, callback) {
-  // Silently drop messages after the session is closed
-  if (this.client.state === 'closed') return;
-  this.client.send(chunk);
-  callback();
-};
-
-ClientStream.prototype._stopClient = function() {
-  var client = this.client;
-  client.stop(function() {
-    client.close();
-  });
-};
+  _stopClient() {
+    this.client.stop(function() {
+      client.close();
```

Comment:
This'll fail because `client` isn't a variable, should keep `var client = this.client`, or alternatively use an arrow function and `this.client`